src/Jackalope/Transport/Jackrabbit/Client.php

```diff
@@ -1328,6 +1333,11 @@ protected function isStringValid($string)
     {
         $regex = '/[^\x{9}\x{a}\x{d}\x{20}-\x{D7FF}\x{E000}-\x{FFFD}]+/u';
 
+        if ($this->version && version_compare($this->version, '2.18.0', '>=')) {
```

Comment:
please extract this version into a constant declared next to the VERSION constant. lets call it UTF8_SUPPORT_MINIMAL_VERSION or something like that.